src/modules/billing/jobs/create-charge-complete.js

```diff
@@ -93,29 +98,20 @@ const finaliseReadyBatch = async (job, messageQueue) => {
 
 const actions = {
   [batchStatuses.processing]: doNothing,
+  [batchStatuses.error]: doNothing,
   [batchStatuses.empty]: finaliseEmptyBatch,
   [batchStatuses.ready]: finaliseReadyBatch
 };
 
-const handleCreateChargeComplete = async (job, messageQueue) => {
-  batchJob.logOnComplete(job);
-
-  if (batchJob.hasJobFailed(job)) {
-    return batchJob.failBatch(job, messageQueue, BATCH_ERROR_CODE.failedToCreateCharge);
-  }
-
-  const { batchId } = parseJob(job);
-
+const handleCreateChargeComplete = async (job, messageQueue, batch) => {
   try {
-    const status = await getBatchProgressStatus(batchId);
-
+    const status = await getBatchProgressStatus(batch.id);
     await actions[status](job, messageQueue);
   } catch (err) {
     batchJob.logOnCompleteError(job);
-    await batchService.setErrorStatus(batchId, BATCH_ERROR_CODE.failedToCreateCharge);
     throw err;
   }
 };
 
-module.exports = handleCreateChargeComplete;
+module.exports = partialRight(createOnCompleteHandler, handleCreateChargeComplete, BATCH_STATUS.processing);
```

Comment:
note to future self: Find out more about `partialRight()`. It seems useful.

Reply:
> Only suggestion (which can be ignored) is that it might be worth putting batchService.setErrorStatus inside of batchJob.logHandlingError just for the sake of brevity.

Done now thanks